src/server/index_patterns/service/lib/__tests__/resolve_time_pattern.js

```diff
@@ -0,0 +1,99 @@
+import sinon from 'sinon';
+import expect from 'expect.js';
+import { noop } from 'lodash';
+
+import { callIndexAliasApi } from '../es_api';
+import * as callIndexAliasApiNS from '../es_api';
+import { timePatternToWildcard } from '../time_pattern_to_wildcard';
+import * as timePatternToWildcardNS from '../time_pattern_to_wildcard';
+
+import { resolveTimePattern } from '../resolve_time_pattern';
+
+const TIME_PATTERN = '[logs-]dddd-YYYY.w';
+
+describe('server/index_patterns/service/lib/resolve_time_pattern', () => {
+  let sandbox;
+  beforeEach(() => sandbox = sinon.sandbox.create());
+  afterEach(() => sandbox.restore());
+
+  describe('resolveTimePattern()', () => {
+    describe('pre request', () => {
+      it('uses callIndexAliasApi() fn', async () => {
+        sandbox.stub(callIndexAliasApiNS, 'callIndexAliasApi').returns({});
+        await resolveTimePattern(noop, TIME_PATTERN);
+        sinon.assert.calledOnce(callIndexAliasApi);
+      });
+
+      it('converts the time pattern to a wildcard with timePatternToWildcard', async () => {
+        const timePattern = {};
+        const wildcard = {};
+
+        sandbox.stub(timePatternToWildcardNS, 'timePatternToWildcard')
+          .returns(wildcard);
+
+        await resolveTimePattern(noop, timePattern);
+        sinon.assert.calledOnce(timePatternToWildcard);
+        expect(timePatternToWildcard.args[0][0]).to.be(timePattern);
```

Comment:
I might be remembering this wrong, but I think this could be:

```
expect(timePatternToWildcard.firstCall.args).to.eql([timePattern]);
```

(I think it reads a little better, and it will ensure all the args are covered in the test)